Matter_Engine/Compute/AggregateVolumetricMaterialTakeoff.cs

```diff
@@ -0,0 +1,98 @@
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using BH.oM.Physical.Constructions;
+using BH.oM.Physical.Materials;
+
+using BH.oM.Base.Attributes;
+using System.ComponentModel;
+using BH.oM.Physical.Elements;
+using BH.oM.Geometry;
+using BH.oM.Dimensional;
+using BH.oM.Quantities.Attributes;
+using BH.Engine.Base.Objects;
+using BH.oM.Base;
+using BH.Engine.Diffing;
+using BH.Engine.Base;
+using BH.oM.Diffing;
+
+namespace BH.Engine.Matter
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Calculates an aggregate VolumetricMaterialTakeoff from a collection of elements.")]
+        [Input("elements", "The elements to iterate over in generation of the VolumetricMaterialTakeoff.")]
+        [Input("checkForTakeoffFragment", "If true and the provided element is a BHoMObject, the incoming item is checked if it has a VolumetricMaterialTakeoff fragment attached, and if so, returns it. If false, the VolumetricMaterialTakeoff returned will be calculated, independant of fragment attached.")]
+        [Output("volumetricMaterialTakeoff", "A VolumetricMaterialTakeoff containing the unique materials across all elements.")]
+        public static VolumetricMaterialTakeoff AggregateVolumetricMaterialTakeoff(IEnumerable<IElementM> elements, bool checkForTakeoffFragment = false)
+        {
+            return AggregateVolumetricMaterialTakeoff(elements.Select(x => x.IVolumetricMaterialTakeoff(checkForTakeoffFragment)));
+        }
+
+        /***************************************************/
+
+        [Description("Calculates an aggregate VolumetricMaterialTakeoff from a collection individual VolumetricMaterialTakeoffs.")]
+        [Input("volumetricMaterialTakeoffs", "The individual VolumetricMaterialTakeoffs to aggregate together.")]
+        [Output("volumetricMaterialTakeoff", "A VolumetricMaterialTakeoff incorporating the provided materials and volumes from each individual VolumetricMaterialTakeoff.")]
+        public static VolumetricMaterialTakeoff AggregateVolumetricMaterialTakeoff(IEnumerable<VolumetricMaterialTakeoff> volumetricMaterialTakeoffs)
+        {
+            List<VolumetricMaterialTakeoff> localMatTakeoffs = volumetricMaterialTakeoffs.ToList();
+
+            Dictionary<string, Tuple<Material, double>> hashedMaterialVolumeTuples = new Dictionary<string, Tuple<Material, double>>();
+
+            for (int j = 0; j < localMatTakeoffs.Count; j++)
+            {
+                for (int i = 0; i < localMatTakeoffs[j].Materials.Count; i++)
+                {
+                    Material mat = localMatTakeoffs[j].Materials[i];
+                    double volume = localMatTakeoffs[j].Volumes[i];
+                    string hash = mat.Hash();
```

Comment:
Because we're using the Hash here, would it be ever useful to expose the `BaseComparisonConfig` in the method inputs, in case we want to programmatically set configurations on how it is calculated?

Reply:
Actually, good point. Will amend